### ISSUE TITLE
Slack bot not allowed to create files channel; unhelpful error message when it fails

### ISSUE THREAD

**Issue by j005u** [metabase/metabase#3949]
Metabot responds fine to the help and list commands however show triggers the error "Slack API error: user_is_bot".

```
12-14 12:59:26 WARN integrations.slack :: {:message "Slack API error: user_is_bot", :response {:ok false, :error "user_is_bot"}}

12-14 12:59:26 ERROR metabase.metabot :: [
"integrations.slack$handle_response.invokeStatic(slack.clj:29)"
"integrations.slack$handle_response.invoke(slack.clj:21)"
"integrations.slack$do_slack_request.invokeStatic(slack.clj:33)"
"integrations.slack$do_slack_request.doInvoke(slack.clj:31)"
"integrations.slack$create_files_channel_BANG_.invokeStatic(slack.clj:59)"
"integrations.slack$create_files_channel_BANG_.invoke(slack.clj:56)"
"integrations.slack$get_or_create_files_channel_BANG_.invokeStatic(slack.clj:78)"
"integrations.slack$get_or_create_files_channel_BANG_.invoke(slack.clj:74)"
"pulse$create_and_upload_slack_attachments_BANG_.invokeStatic(pulse.clj:47)"
"pulse$create_and_upload_slack_attachments_BANG_.invoke(pulse.clj:44)"
"metabot$show$fn__29927.invoke(metabot.clj:129)"]

```

I'm running Metabase 0.20.3.
I created the API key for my bot through the provided link in the settings panel.

Any tips?

**Comment by camsaul** [comment]
@j005u did you [create the `metabase_files` channel in Slack when setting up the MetaBot](http://www.metabase.com/docs/latest/administration-guide/08-setting-up-slack.html)?

[That's the Slack API error message you'll see if a bot user tries to create it -- bots aren't allowed to create channels](https://api.slack.com/bot-users#).

@metabase/core-developers @salsakran If we're asking people to use a bot token and manually create `metabase_files`, why did we leave in the code that tries to automatically create it (knowing it will fail)? We should probably remove that code and instead just throw an Exception with a better error message if the channel doesn't exist.

**Comment by j005u** [comment]
Oh my, this is embarrassing. I have no idea how I managed to miss that while triple checking the docs for any additional setup instructions. It's even right there on the admin page.

Yes, the error message wasn't super helpful, but still totally a case of user error.

**Comment by hanksudo** [comment]
@j005u Did you solve the issue? I am facing the same error.

**Comment by buraks** [comment]
Any update on this? I am facing the same issue on v0.21.1

**Comment by camsaul** [comment]
@hanksudo just create the files channel manually as mentioned in the instructions

**Comment by calz1** [comment]
If anyone else gets this error and the channel _exists_, my coworker figured out the channel must be **public**. If you made it private, rename the private one to something else and create a new #metabase_files.

**Comment by olalonde** [comment]
That's a bug though... bot should be able to support a private #metabase_files channel, especially when he was invited in it :(

**Comment by flamber** [comment]
There was a lot of fixes in 0.36.x, including private channel support #2694 (latest release is currently 0.41.1): https://github.com/metabase/metabase/releases/latest

This issue hasn't been reported for several years, so closing.
And the entire Slack implementation will be rewritten in the future - see #12383

If you are still experiencing this issue, then please include "Diagnostic Info" from Admin > Troubleshooting, and the full stacktrace from Admin > Troubleshooting > Logs, and any other helpful information (like which Slack commands you're using etc).
